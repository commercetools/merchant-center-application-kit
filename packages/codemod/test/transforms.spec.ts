/* eslint-disable import/first */
jest.autoMockOff();

import fs from 'fs';
import path from 'path';
// @ts-ignore
import { defineSnapshotTest } from 'jscodeshift/dist/testUtils';

const fixturesPath = path.join(__dirname, 'fixtures');

describe.each`
  transformName                            | fixtureName
  ${'remove-deprecated-modal-level-props'} | ${'remove-deprecated-modal-level-props.tsx'}
`('testing transform "$transformName"', ({ transformName, fixtureName }) => {
  // Assumes transform is one level up from __tests__ directory
  const module = require(path.join(
    __dirname,
    '../src/transforms',
    transformName
  ));
  const inputPath = path.join(fixturesPath, fixtureName);
  const source = fs.readFileSync(inputPath, 'utf8');
  defineSnapshotTest(module, null, source);
});
