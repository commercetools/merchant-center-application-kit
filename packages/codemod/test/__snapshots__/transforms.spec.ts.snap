// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing transform "remove-deprecated-modal-level-props" transforms correctly 1`] = `
"import {
  InfoModalPage,
  FormModalPage,
  CustomFormModalPage,
  TabularModalPage,
  TabHeader,
} from '@commercetools-frontend/application-components';

function Modal() {
  return <>
    <InfoModalPage isOpen title=\\"Hello\\" onClose={() => {}}>
      Hello
    </InfoModalPage>
    <FormModalPage
      isOpen
      title=\\"Hello\\"
      onPrimaryButtonClick={() => {}}
      onSecondaryButtonClick={() => {}}
      onClose={() => {}}>
      Hello
    </FormModalPage>
    <CustomFormModalPage isOpen title=\\"Hello\\" onClose={() => {}}>
      Hello
    </CustomFormModalPage>
    <TabularModalPage
      isOpen
      title=\\"Hello\\"
      tabControls={
        <>
          <TabHeader to=\\"/one\\" label=\\"One\\" />
          <TabHeader to=\\"/two\\" label=\\"Two\\" />
        </>
      }
      onClose={() => {}}>
      Hello
    </TabularModalPage>
  </>;
}

export default Modal;"
`;
